### PR TITLE
Fix wrong support of USR_VIMRC_FILE4.

### PR DESCRIPTION
**Pull request by Yamakaky**
It was sourced, but it was not treated as USR_VIMRC[23]?.

### CODE REVIEW

**Comment by Pyrohh** [comment]
Please see [CONTRIBUTING.md#submitting-contributions](https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#submitting-contributions).

**Comment by Yamakaky** [comment]
Done for the CLA. What should have I done before submitting ?

**Comment by Pyrohh** [comment]
> What should have I done before submitting ?

Everything else should be mentioned in the link above. The main issue seems to be that you're working on the master branch as opposed to the feature branch; in practice, I'm not sure what difference it makes though.

**Comment by Yamakaky** [comment]
As it's a straightforward one-commit PR, I thought it would be OK.

**Comment by Pyrohh** [comment]
It probably is, I suppose I just want to treat people equally in their PRs, so some people don't feel like they're undergoing more bureaucratic things than others. Anyways, I'll take a look.

**Comment by Yamakaky** [comment]
I understand.

**Comment by Pyrohh** [comment]
LGTM

**Comment by glts** [comment]
Hold on. Do we support any of the systems that have four user vimrc locations (`:h vimrc`)? I don't think so (Amiga!).

It might be best to remove all code to do with the fourth user vimrc instead.

**Comment by Pyrohh** [comment]
> Hold on. Do we support any of the systems that have four user vimrc locations (:h vimrc)? I don't think so (Amiga!).
> 
> It might be best to remove all code to do with the fourth user vimrc instead.

I think you're right, let's see if anyone else chimes in.

**Comment by Yamakaky** [comment]
I'm working on something that may need it, please let it for a few days.

**Comment by splinterofchaos** [comment]
> The main issue seems to be that you're working on the master branch as opposed to the feature branch; in practice, I'm not sure what difference it makes though.

It can cause problems/confusions if one wants to create another PR. It can be a slow process refining a PR, and one might want to make further unrelated changes later. There's really no technical reason, but two PRs would require at least two branches.

Briefly scanning these changes, they look good if we support `USR_VIMRC_FILE4`, but I notice that we don't define `*_FILE4` or `3` anywhere, so this won't actually do anything. Should we define them? If so, to what?

**Comment by Yamakaky** [comment]
No merge ?

**Comment by splinterofchaos** [comment]
As I said, this PR doesn't actually do anything because we don't define `USR_VIMRC_FILE4` or even `USR_VIMRC_FILE3`. We need to answer some questions: What do we set them to? Is this something for users to configure, and if so, how? What if we just decide to erase all references to `USR_VIMRC_FILR[34]` and not support them? What's the use case?

Now, I don't mean to block this PR because it would certainly be safe to merge. It's just that I haven't seen anyone propose answers to these questions and that gives me pause. If we leave this be and decide on the issue later, you can always amend the commit or someone else can cherry pick it.

Basically: LGTM :+1:, but I'll leave the decision of whether to merge to someone else.

**Comment by Yamakaky** [comment]
I put the commit on his own branch. Now I understand the hard way why I should have done it the first time...

**Comment by Pyrohh** [comment]
@splinterofchaos From the vim source tree:

```
#endif
#ifndef USR_VIMRC_FILE3
# define USR_VIMRC_FILE3 "home:vimfiles:vimrc"
#endif
#ifndef USR_VIMRC_FILE4
# define USR_VIMRC_FILE4 "$VIM/.vimrc"
#endif
```

And `:h vimrc` in vanilla vim:
```
                Unix            $HOME/.vimrc or $HOME/.vim/vimrc
                OS/2            $HOME/.vimrc, $HOME/vimfiles/vimrc
                                or $VIM/.vimrc (or _vimrc)
                MS-Windows      $HOME/_vimrc, $HOME/vimfiles/vimrc
                                or $VIM/_vimrc
                Amiga           s:.vimrc, home:.vimrc, home:vimfiles:vimrc
                                or $VIM/.vimrc
```

Given that, they should probably be removed.

**Comment by Pyrohh** [comment]
Those defines are in `os_{amiga,unix,dos}.h` by the way. The ones in `os_unix.h` are ifdef'd behind `OS2` and `VMS`.
